app/controllers/users_controller.rb

```diff
@@ -22,9 +22,47 @@ def create
     end
   end
 
+  def request_reset
+  end
+
+  def send_reset_link
+    user = User.where(email: params[:email]).first
+    if user
+      scheme = request.env['rack.url_scheme']
+      host = request.env['HTTP_HOST']
+      link = user.generate_reset_link(scheme, host)

```

Comment:
Any chance you could show how or which one to use? I tried to use it inside a model but couldn't make it access environment. I feel rails must have something simple and elegant for this, but what?